frontend/app-development/features/processEditor/handlers/OnProcessTaskRemoveHandler.test.ts

```diff
@@ -54,14 +112,68 @@ describe('OnProcessTaskRemoveHandler', () => {
       } as TaskEvent,
     };
 
-    const onProcessTaskRemoveHandler = new OnProcessTaskRemoveHandler(
-      org,
-      app,
+    const onProcessTaskRemoveHandler = createOnRemoveProcessTaskHandler({
       currentPolicy,
-      mutateApplicationPolicy,
-    );
+    });
     onProcessTaskRemoveHandler.handleOnProcessTaskRemove(taskMetadata);
 
-    expect(mutateApplicationPolicy).toHaveBeenCalledWith(expectedResponse);
+    expect(mutateApplicationPolicyMock).toHaveBeenCalledWith(expectedResponse);
+    expect(deletelayoutSetMock).not.toHaveBeenCalled();
+  });
+
+  it('should delete layoutSet for payment-task if layoutSet exists', () => {
+    const layoutSets: LayoutSets = {
+      sets: [{ id: 'testLayoutSetId', dataType: 'payment', tasks: ['testElementId'] }],
+    };
+
+    const taskMetadata: OnProcessTaskEvent = {
```

Comment:
Maybe we can refactor this out in a function `createTaskMetadata = (taskType: TaskType) => { ... }`? 😄 
I can see the same block is used several places with only the `taskType` being different.